client/executor/src/native_executor.rs

```diff
@@ -170,7 +176,7 @@ where
 	/// runtime is invalidated on any `panic!` to prevent a poisoned state. `ext` is already
 	/// implicitly handled as unwind safe, as we store it in a global variable while executing the
 	/// native runtime.
-	fn with_instance<R, F>(
+	pub fn with_instance<R, F>(
```

Comment:
```suggestion
	fn with_instance<R, F>(
```

Reply:
To be honest, I think this method should be exported.

Reply:
Why?

Reply:
Because `CodeExecutor::call` is not generic. I means it use `preregister_builtin_ext`

Reply:
Can you please write down exactly on what you want. Why do you need to have this exported. Please write some reasoning. And not just "I means it use preregister_builtin_ext". While this is right, it doesn't explain on what you try to do and fail with the current implementation.

Reply:
This `call` in `CodeExecutor` implemented by `WasmExecutor` maybe need to be reimplmented in the future. So we need  to `with_instance` to be `pub`.

Reply:
> This `call` in `CodeExecutor` implemented by `WasmExecutor` maybe need to be reimplmented in the future.

Why does it needs to be reimplemented?

Reply:
Seems to say it again :) because we want to remove some host functions.

Reply:
Instead, I wonder why these things don't need to be exposed, 
e.g. I think `RuntimeCache` is also very useful.

Reply:
> e.g. I think `RuntimeCache` is also very useful.

FYI, that type will be most likely removed in the refactoring anyway in favor of a more holistic approach.